spec/models/reports/patient_follow_up_spec.rb

```diff
@@ -0,0 +1,210 @@
+    follow_ups = described_class.where(patient: patient, month_string: june_2021[:month_string], facility: facility)
+    expect(follow_ups.map(&:user)).to contain_exactly(user, another_user)
+  end
+
+  it "contains separate records for distinct facilities" do
+    patient = create(:patient, recorded_at: june_2021[:long_ago])
+    user = create(:user)
+    facility = create(:facility)
+    another_facility = create(:facility)
+
+    create(
+      :blood_pressure,
+      patient: patient,
+      user: user,
+      facility: facility,
+      recorded_at: june_2021[:now]
+    )
+
+    create(
+      :blood_pressure,
+      patient: patient,
+      user: user,
+      facility: another_facility,
+      recorded_at: june_2021[:now]
+    )
+
+    RefreshReportingViews.call
+
+    expect(described_class.count).to eq(2)
+    follow_ups = described_class.where(patient: patient, month_string: june_2021[:month_string], user: user)
+    expect(follow_ups.map(&:facility)).to contain_exactly(facility, another_facility)
+  end
+
+  it "contains separate records for distinct patients" do
+    patient = create(:patient, recorded_at: june_2021[:long_ago])
+    another_patient = create(:patient, recorded_at: june_2021[:long_ago])
+    user = create(:user)
+    facility = create(:facility)
+
+    create(
+      :blood_pressure,
+      patient: patient,
+      user: user,
+      facility: facility,
+      recorded_at: june_2021[:now]
+    )
+
+    create(
+      :blood_pressure,
+      patient: another_patient,
+      user: user,
+      facility: facility,
+      recorded_at: june_2021[:now]
+    )
+
+    RefreshReportingViews.call
+
+    expect(described_class.count).to eq(2)
+    follow_ups = described_class.where(month_string: june_2021[:month_string], facility: facility, user: user)
+    expect(follow_ups.map(&:patient)).to contain_exactly(patient, another_patient)
+  end
+
+  it "does not count activity in the registration month" do
+    patient = create(:patient, recorded_at: june_2021[:beginning_of_month])
+    user = create(:user)
+    facility = create(:facility)
+
+    create(
+      :blood_pressure,
+      patient: patient,
+      user: user,
+      facility: facility,
+      recorded_at: june_2021[:end_of_month]
+    )
+
+    RefreshReportingViews.call
+
+    expect(described_class.count).to eq(0)
+  end
+
+  it "does not count more than one record per month for the same patient, facility, and user" do
+    patient = create(:patient, recorded_at: june_2021[:long_ago])
+    user = create(:user)
+    facility = create(:facility)
+
+    create(
+      :blood_pressure,
+      patient: patient,
+      user: user,
+      facility: facility,
+      recorded_at: june_2021[:beginning_of_month]
+    )
+
+    create(
+      :blood_pressure,
+      patient: patient,
+      user: user,
+      facility: facility,
+      recorded_at: june_2021[:end_of_month]
+    )
+
+    RefreshReportingViews.call
+
+    expect(described_class.count).to eq(1)
```

Comment:
Is there anything to assert on here to make sure that the follow up captured is from the `end_of_month` BP - to make sure it grabs the most recent record?

Reply:
There actually isn't anything that ties the record to the end-of-month BP, since we're only storing a month date on each record. However, this assertion is still pretty light. I'll add another assertion that the one record we see in here has the right attributes on it.